gslib/command.py

```diff
@@ -1929,6 +1953,61 @@ def _ApplyThreads(self, thread_count, process_count, recursive_apply_level,
         # block the semaphore on a WorkerThread completion.
         worker_semaphore.release()
 
+  def _GetGcloudStorageArgs(self, sub_opts, gsutil_args, gcloud_storage_map):
+    if gcloud_storage_map is None:
+      raise GcloudStorageTranslationError(
+          'Command "{}" cannot be translated to gcloud storage because the'
+          ' translation mapping is missing.'.format(self.command_name))
+    args = []
+    if isinstance(gcloud_storage_map.gcloud_command, str):
+      args.append(gcloud_storage_map.gcloud_command)
+    elif isinstance(gcloud_storage_map.gcloud_command, dict):
+      # If a command has sub-commands, e.g pap set, pap get.
+      # All the flags mapping must be present in the subcommand's map
+      # because gsutil does not have command specific flags
+      # if sub-commands are present.
+      if gcloud_storage_map.flag_map:
+        raise ValueError(
+            'Flags mapping found at command lavel for the command: {}.'.format(
+                self.command_name))
+      sub_command = gsutil_args[0]
+      sub_opts, parsed_args = self.ParseSubOpts(args=gsutil_args[1:],
+                                                update_sub_opts_and_args=False)
+      return self._GetGcloudStorageArgs(
+          sub_opts, parsed_args,
+          gcloud_storage_map.gcloud_command.get(sub_command))
+    else:
+      raise ValueError('Incorrect mapping found for "{}" command'.format(
+          self.command_name))
+
+    if sub_opts:
+      for option, value in sub_opts:
+        if option not in gcloud_storage_map.flag_map:
+          raise GcloudStorageTranslationError(
+              'Command option "{}" cannot be translated to'
+              ' gcloud storage'.format(option))
+        args.append(gcloud_storage_map.flag_map[option].gcloud_flag)
+        if value != '':
+          args.append(value)
+    return args + gsutil_args
+
+  def GetGcloudStorageArgs(self):
+    """Translates the gsutil command flags to gcloud storage flags.
+
+    It uses the command_spec.gcloud_storage_map field which provides the
```

Comment:
In this type of sentence, "which" is for optional info and "that" is for mandatory info.

Reply:
Fixed. That's good to know. Thanks for explaining it.